test/Microsoft.IdentityModel.Protocols.OpenIdConnect.Tests/ConfigurationManagerTests.cs

```diff
@@ -510,6 +510,48 @@ public static TheoryData<ConfigurationManagerTheoryData<OpenIdConnectConfigurati
             }
         }
 
+        [Fact]
+        public async Task CheckSyncAfter()
+        {
+            // This test checks that the _syncAfter field is set correctly after a refresh.
+            var context = new CompareContext($"{this}.CheckSyncAfter");
+
+            var docRetriever = new FileDocumentRetriever();
+            var configManager = new ConfigurationManager<OpenIdConnectConfiguration>("OpenIdConnectMetadata.json", new OpenIdConnectConfigurationRetriever(), docRetriever);
+
+            // This is the minimum time that should pass before an automatic refresh occurs
+            // stored in advance to avoid any time drift issues.
+            DateTimeOffset minimumRefreshInterval = DateTimeOffset.UtcNow + configManager.AutomaticRefreshInterval;
+
+            // get the first configuration, internal _syncAfter should be set to a time greater than UtcNow + AutomaticRefreshInterval.
+            var configuration = await configManager.GetConfigurationAsync(CancellationToken.None);
+
+            // force a refresh by setting internal field
+            TestUtilities.SetField(configManager, "_syncAfter", DateTimeOffset.UtcNow - TimeSpan.FromHours(1));
+            configuration = await configManager.GetConfigurationAsync(CancellationToken.None);
+            // wait 100ms here because update of config is run as a new task.
```

Comment:
This sounds flaky. Is there a way to store a reference to the fire-and-forget `Task` and expose it only for tests?

Reply:
@msbw2 this seems to work.
Can you provide an example of what you would like to see?

Reply:
I'm thinking of an `internal` property `UpdateCurrentConfigurationTask` which is only used in tests to `await` the task to avoid sleeping and hoping the task isn't delayed for some reason. I would consider this a non-blocker, though.

Reply:
I think this might be why this test is failing. You set `_syncAfter` above, and the failure message indicates that that's the value when you retrieve it below. It seems 100ms isn't long enough for the task to be reliably run, but I'd advise against increasing it experiementally.